src/Microsoft.Diagnostics.Monitoring.WebApi/Controllers/DiagController.cs

```diff
@@ -636,29 +634,21 @@ private string GetDiagnosticPortName()
             TimeSpan duration,
             string egressProvider)
         {
-            string fileName = TraceUtilities.GenerateTraceFileName(processInfo.EndpointInfo);
+            IArtifactOperation traceOperation = _traceOperationFactory.Create(
+                processInfo.EndpointInfo,
+                configuration,
+                duration);
+
+            if (_diagnosticPortOptions.Value.ConnectionMode == DiagnosticPortConnectionMode.Listen)
+            {
+                IDisposable operationRegistration = _operationTrackerService.Register(processInfo.EndpointInfo);
+                HttpContext.Response.RegisterForDispose(operationRegistration);
```

Comment:
nit: This changes the scope of when this registration is disposed (it was created and disposed in the same function previously). This is probably okay so long as we are guaranteed that this will be disposed, even in the case of something catastrophic happening when handling the http request.